pkg/iam/nodes_template.go

```diff
@@ -129,6 +129,14 @@ const nodesTemplate = `{
                 "*"
             ],
             "Effect": "Allow"
+        },
+        {
+            "Action": [
+                "secretsmanager:GetSecretValue",
+                "secretsmanager:DeleteSecret"
+            ],
+            "Resource": "arn:*:secretsmanager:*:*:secret:aws.cluster.x-k8s.io/*",
+            "Effect": "Allow"
```

Comment:
looks like worker nodes don't need it for now, but just adding to have it for future